File: src/main/java/ru/ioleksiv/telegram/bot/core/model/telegram/method/group/GetChatMember.java
package ru.ioleksiv.telegram.bot.core.model.telegram.method.group;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.ioleksiv.telegram.bot.core.controller.network.Networker;
import ru.ioleksiv.telegram.bot.core.model.telegram.method.interfaces.ChatAction;
import ru.ioleksiv.telegram.bot.core.model.telegram.objects.type.ChatMember;
import ru.ioleksiv.telegram.bot.core.model.telegram.responses.CommonResponse;
import ru.ioleksiv.telegram.bot.core.model.telegram.responses.ResponseCollection;

/**
 * @see <a href="https://core.telegram.org/bots/api#getchatmember>getChatMember</a>
 */
public class GetChatMember  extends ChatAction<ChatMember > {
    private static final String METHOD = "getChatMember";

    /**
     *user_id	Integer	Yes	Unique identifier of the target user
     */
    @JsonProperty("user_id")
    private Long userId = null;

    protected GetChatMember(Networker networker) {
        super(METHOD, networker);
    }

    public Long getUserId() {
        return userId;
    }

    @Override
    public Class<? extends CommonResponse<ChatMember>> getResultWrapperClass() {
        return ResponseCollection.ChatMemberResponse.class;
    }

    public GetChatMember  setUserId(Long userId) {
        this.userId = userId;
        return this;
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/method/group/GetChatMember.kt
package io.github.oleksivio.tl.kbot.server.api.method.group

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.model.ApiDict
import io.github.oleksivio.tl.kbot.server.api.model.ChatMemberResponse
import io.github.oleksivio.tl.kbot.server.api.model.method.common.ChatAction
import io.github.oleksivio.tl.kbot.server.api.objects.ChatId
import io.github.oleksivio.tl.kbot.server.api.objects.std.ChatMember

/**
 * @see [getChatMember](https://core.telegram.org/bots/api/#getchatmember)
 */
data class GetChatMember(
    /**
     * chat_id Integer or String Yes Unique identifier for the target chat or username of the target channel
     */
    @JsonProperty(ApiDict.CHAT_ID_KEY)
    override val chatId: ChatId,
    /**
     * user_id Integer Unique identifier of the target user
     */
    @JsonProperty(ApiDict.USER_ID_KEY)
    val userId: Long
) : ChatAction<ChatMember>() {

    @JsonIgnore
    override val resultWrapperClass = ChatMemberResponse::class

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method = "getChatMember"
}
